staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go

```diff
@@ -378,3 +385,186 @@ func (o *ExposeServiceOptions) RunExpose(cmd *cobra.Command, args []string) erro
 	}
 	return nil
 }
+
+func (o *ExposeServiceOptions) createService() (*corev1.Service, error) {
+	if len(o.Selector) == 0 {
+		return nil, fmt.Errorf("selector must be specified")
+	}
+	selector, err := parseLabels(o.Selector)
+	if err != nil {
+		return nil, err
+	}
+
+	var labels map[string]string
+	if len(o.Labels) > 0 {
+		labels, err = parseLabels(o.Labels)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	name := o.Name
+	if len(name) == 0 {
+		name = o.DefaultName
+		if len(name) == 0 {
+			return nil, fmt.Errorf("name must be specified")
+		}
+	}
+
+	var portProtocolMap map[string]string
+	if o.Protocols != "" {
```

Comment:
Move this to `Validate`

Reply:
`default-name` cannot be moved to `Validate` because if user does not provide a name it will call the API server and get from the `info` object which mean user does not have to provide a name beforehand. 
https://github.com/kubernetes/kubectl/blob/1bf219da9bdd8c03758195f5b43f1ddfbe31c3e7/pkg/cmd/expose/expose.go#L254